### ISSUE TITLE
database crash with lost data

### ISSUE THREAD

**Issue by beckettsean** [influxdb/influxdb#3066]
I will try to find a simpler repro case but I'm documenting what I have for now.

0.9.0 final, installed via Homebrew, stock install with no alterations. I'm pretty sure I didn't even generate a new config file, I just launched influxd with `influxd -config /usr/local/etc/influxdb.conf 2>> ~/workspace/logs/influxd.log &`

Today I've been recreating various reported issues. I've panicked my database a half-dozen times or more. Each time it panics I just re-run the above command to relaunch the daemon. I never exit the CLI, I just keep it open in a different Terminal window. 

I was recreating the panic issue with `PERCENTILE` (#3065) when my data disappeared. Here's the unedited CLI output:

```
> use mydb
Using database mydb
> show measurements
name: measurements
------------------
name
amount
boolean
foobar
meas
negative
tag_key_query
test-show-tags
time

> show measurements
name: measurements
------------------
name
amount
boolean
cpu_load_short
foobar
meas
negative
tag_key_query
test-show-tags
time

> select * from foobar
name: foobar
tags: key=val1
time				value
----				-----
2015-06-18T19:09:52.947299286Z	1
2015-06-18T19:10:01.619062526Z	2
2015-06-18T19:10:03.171388731Z	3


name: foobar
tags: key=val2
time				value
----				-----
2015-06-18T19:10:06.339544273Z	3
2015-06-18T19:10:07.611584869Z	4
2015-06-18T19:10:08.763866774Z	5

> select percentile(value, 30) from foobar
ERR: Get http://localhost:8086/query?db=mydb&q=select+percentile%28value%2C+30%29+from+foobar: EOF
```

After this query my db panicked (see #3065). I wondered if the panic was because the field values in `foobar` are all integers, so I went looking for a measurement with floats. Immediately after the above error my CLI output is as follows:

```
> select * from negative
ERR: measurement not found: "mydb"."default".negative
> show measurements
name: measurements
------------------
name
cpu_load_short

> use mydb
Using database mydb
> show measurements
name: measurements
------------------
name
cpu_load_short

> show databases
name: databases
---------------
name
mydb
db
tag_issue
aaa

> use db
Using database db
> show measurements
> use aaa
Using database aaa
> show measurements
name: measurements
------------------
name
df

> use mydb
Using database mydb
> show measurements
name: measurements
------------------
name
cpu_load_short

> use tag_issue
Using database tag_issue
> show measurements
> use aaa
Using database aaa
> show measurements
name: measurements
------------------
name
df

> use mydb
Using database mydb
> show measurements
name: measurements
------------------
name
cpu_load_short

> show measurements
name: measurements
------------------
name
cpu_load_short

> show retention policies mydb
name	duration	replicaN	default
default	0		1		true

> show series
name: cpu_load_short
--------------------
_key								direction	host		region
cpu_load_short,direction=in,host=server01,region=us-west	in		server01	us-west
cpu_load_short,host=server02,region=us-west					server02	us-west
```

Notice the stark difference in `show measurements` before and after the db panic. `mydb` used to have nine measurements, now it has one, and that one is the most-recently created measurement. The other databases appear unaffected, but I cannot find any other series in `mydb` except the last one made.

I tar'd up my database files from `/usr/local/var/influxdb/`, my conf file from `/usr/local/etc/`, and my log file, and uploaded them as `influxd-2015-06-19.missing_measurements.tgz` to Dropbox: https://www.dropbox.com/s/cjn2d2h6q1uf9dr/influxd-2015-06-19.missing_measurements.tgz?dl=0.

I'm going to wipe out my installation, re-install, and try to reproduce.

**Comment by beckettsean** [comment]
I tried to recreate the situation and failed. After wiping the machine I did the following:
- `brew install influxdb`
- `$ influxd -config /usr/local/etc/influxdb.conf`
- made two measurements with dummy data (repeated `insert foo,t1=k1 value=1000`, `insert bar value=1`, etc.)
- recreated the `PERCENTILE` panic as described in https://github.com/influxdb/influxdb/issues/3065#issuecomment-113669589 
- `show measurements` still shows all four existing measurements, as expected.

Since the #3065 crash is not the immediate cause I'm at a loss as to how to recreate the issue. I'm happy to experiment if anyone else has a suggestion.
